### PR TITLE
Brooms Upgraded: Disposal Bin Edition

### PR DESCRIPTION
**Pull request by bobbah**
## About The Pull Request
Allows the broom to sweep items into a disposal bin. Also allows brooms to be made in autolathe / service protolathe.

https://www.youtube.com/watch?v=xSqE7hUJ3EE

also at the request of MSO the broom has been renamed push broom

## Why It's Good For The Game
It'll really _clean up_ our use of brooms.

## Changelog
:cl: bobbahbrown
add: Nanotrasen has recently retrofitted all disposal bins with built-in vacuum units, allowing brooms to sweep garbage into them!
add: Push brooms can now be manufactured in the service protolathe and any autolathe.
/:cl:

### CODE REVIEW

**Comment by JJRcop** [comment]
I think this makes the job too easy. Janitors do have a trash bag they can use to pick up the items.

**Comment by bobbahbrown** [comment]
> I think this makes the job too easy. Janitors do have a trash bag they can use to pick up the items.

comparing using a garbage bag to pick up items in front of the bin to put inside against simply brooming them in sounds like compelling gameplay that is truly riveting and challenging

**Comment by Nickkoranda** [comment]
Trash bags can be used to pick up the pieces in rooms you don't have disposals in. This is truly the correct future for janny-kind. <sub>Plus I think actually pushing around trash is more fun that just click every tile with trash</sub>

**Comment by wesoda25** [comment]
powercreep

**Comment by wesoda25** [comment]
wait itd be funnier if i said powersweep

**Comment by Michael-Ashfield** [comment]
Okay broomer

File: code/modules/research/designs/autolathe_designs.dm [hunk 1]
```diff
@@ -20,6 +20,15 @@
 	category = list("initial","Tools","Tool Designs")
 	departmental_flags = DEPARTMENTAL_FLAG_SERVICE
 
+/datum/design/broom
+	name="Push Broom"
```
Comment (nemvar) [inline code/modules/research/designs/autolathe_designs.dm]: Odd name.

Comment (bobbahbrown) [reply code/modules/research/designs/autolathe_designs.dm]: https://en.wiktionary.org/wiki/push_broom

**Comment by MrStonedOne** [comment]
Push broom != broom.

I am not happy you named types generics like broom while making them a speciality like push broom.

A broom where you had to click drag between two adjacent tiles to sweep the debris from one tile to the other would be a fun experiment, hence why push brooms should not have a type call broom, it should be "pushbroom"

**Comment by bobbahbrown** [comment]
> Push broom != broom.
> 
> I am not happy you named the types types generics names like broom while making them a speciality like push broom.
> 
> A broom where you had to click drag between two adjacent tiles to sweep the debris from one tile to the other would be a fun experiment, hence why push brooms should not have a type called broom, it should be "pushbroom"

i didnt name the generics but i can rename them one sec

**Comment by MrStonedOne** [comment]
oh, i had saw the added design datum.

File: code/game/objects/items/twohanded.dm [hunk 2]
```diff
@@ -902,16 +902,24 @@
...
+			else
+				garbage.Move(target_turf, user.dir)
+			i++
+		if(i > 19)
 			break
-	if(i >= 1)
+	if(i > 0)
+		if (target_bin)
+			target_bin.update_icon()
+			to_chat(user, "<span class='notice'>You sweep the pile of garbage into the [target_bin].</span>")
```
Comment (nemvar) [inline code/game/objects/items/twohanded.dm]: Remove the "the". Currently this says "the the garbage bin"

File: code/game/objects/items/twohanded.dm [hunk 3]
```diff
@@ -902,16 +902,24 @@
...
 	var/i = 0
+	var/turf/target_turf = get_step(target, user.dir)
+	var/obj/machinery/disposal/bin/target_bin = locate(/obj/machinery/disposal/bin) in target_turf.contents
 	for(var/obj/item/garbage in target.contents)
 		if(!garbage.anchored)
-			garbage.Move(get_step(target, user.dir), user.dir)
-		i++
-		if(i >= 20)
+			if (target_bin)
+				garbage.forceMove(target_bin)
```
Comment (PKPenguin321) [inline code/game/objects/items/twohanded.dm]: You could make this give a little janitor xp for each piece of garbage

Comment (bobbahbrown) [reply code/game/objects/items/twohanded.dm]: its a slippery slope because it would be ludicrously easy to farm xp in a very naughty way i think

Comment (bobbahbrown) [reply code/game/objects/items/twohanded.dm]: but i do like the idea hmm

File: code/modules/research/designs/autolathe_designs.dm (hunk 1 above)
Comment (Fikou) [reply code/modules/research/designs/autolathe_designs.dm]: this should to in all_nodes.dm i think?

File: code/modules/research/designs/autolathe_designs.dm [hunk 4]
```diff
@@ -20,6 +20,15 @@
...
+	id="pushbroom"
```
Comment (Fikou) [inline code/modules/research/designs/autolathe_designs.dm]: this should to in all_nodes.dm i think?

Comment (bobbahbrown) [reply code/modules/research/designs/autolathe_designs.dm]: not a tech web item

**Comment by bobbahbrown** [comment]
help

**Pull request merged**